hphp/runtime/vm/jit/vasm-arm.cpp

```diff
@@ -1036,6 +1003,15 @@ void lower(Vunit& unit, Inst& inst, Vlabel b, size_t i) {}
 
 ///////////////////////////////////////////////////////////////////////////////
 
+bool allowVreg = true;
```

Comment:
Herp derp, of course—that's the only place where these instructions are meaningful at all...

Okay, let's get the refactoring of these lower loops taken care of first.  There's a good amount of duplication among the different platforms that I've been putting off until the boilerplate started getting out of hand, which appears to be the case now.  Let me put up a diff that should give you what you need to generalize this logic and make it less globally-stateful.  I'll ping here with the diff once I've landed it.